src/app/components/AssetDetail/index.jsx

```diff
@@ -0,0 +1,162 @@
+import ChangePercent from 'Components/ChangePercent'
+import ArrowIcon from 'Components/ArrowIcon'
+
+const getQuery = ({ match }) => match.params.symbol
+
+const marketData = [
+  {
+    title: 'Market Cap',
+    key: 'marketCap',
+  },
+  {
+    title: '24hr Volume',
+    key: 'volume24',
+  },
+  {
+    title: 'Supply',
+    key: 'availableSupply',
+  }
+]
+
+let TO_ASSET = 'BTC'
+let FROM_ASSET = 'ETH'
+
+const AssetDetail = ({ symbol, asset, assetHoldings, isPriceChartLoading }) => {
+  const { name, price, change24, deposit, receive } = asset
+  if (symbol === 'BTC') { TO_ASSET = 'ETH' }
+  if (symbol === 'ETH') { FROM_ASSET = 'BTC' }
+  return (
+    <Layout className='pt-3 p-0 p-sm-3'>
+      <Card>
+      <CardHeader className='grid-group'>
+        <Row className='gutter-3 p-sm-0 p-3 d-flex'>
+          <Col className='d-flex align-items-center pl-sm-4 pl-0 py-2 col-auto' size='sm'>
+            <Media>
+              <Media left>
+                <CoinIcon 
+                  className='mr-2 mt-1' 
+                  symbol={symbol} 
+                  style={{ width: '40px', height: '40px', position: 'relative', top: '2px' }} 
+                  inline
+                /> 
+              </Media>
+              <Media body>
+                <Media className='m-0 mt-1 font-weight-bold' heading>
+                  {name}
+                </Media>
+                <small style={{ position: 'relative', top: '-5px' }} className='text-muted'>[{symbol}]</small>
+              </Media>
+            </Media>
+          </Col>
+          <Col className='col-sm-8 col-md-8 col-lg-2 col-5'>
+            <div className='pl-4 py-2'>
+              <div className='mb-0'>
+                <Units 
+                  className='mt-1 d-inline-block font-weight-bold'
+                  value={price} 
+                  symbol={'$'} 
+                  precision={6} 
+                  prefixSymbol
+                />
+              </div>
+              <small style={{ position: 'relative', top: '-5px' }}><ChangePercent>{change24}</ChangePercent></small>
+              <ArrowIcon
+                style={{ position: 'relative', top: '-5px' }}
+                className={classNames('swapChangeArrow', change24.isZero() ? 'd-none' : null)} 
+                size={.58} dir={change24 < 0 ? 'down' : 'up'} 
+                color={change24 < 0 ? 'danger' : change24 > 0 ? 'success' : null}
+              />
+            </div>
+          </Col>
+          <Col className='col-auto ml-3 mr-3' size='sm'>
+            <div className='py-2'>
+              <div className='text-muted mb-0'>
+                <small>Holdings</small>
+              </div>
+                <Units 
+                  value={assetHoldings || 0} 
+                  symbol={symbol} 
+                  precision={6}
+                />
+            </div>
+          </Col>
+          {marketData.map(({ title, key }, i) => {
+            const flag = (key !== 'availableSupply')
```

Comment:
I would include this as a field in marketData with a better name, `fiat: boolean` perhaps?